test/user.test.js

```diff
@@ -0,0 +1,100 @@
+                password: userData.password
+            })
+            await newUser.save()
+        }
+    })
+    
+    afterEach(async ()=>{
+        await User.deleteMany()
+    })
+
+    afterAll(() => {
+        disconnectDatabase();
+    });
+    describe('GET route', () =>{
+        it('should get all users from database', async () =>{
+            const res = await request.get('/api/users');
+            expect(res.status).toBe(200)
+            expect(res.body.length).toBe(userTestData.length)
+            expect(res.body[0].uuid).toBe(userTestData[0].uuid)
+            expect(res.body[0].userName).toBe(userTestData[0].userName)
+            expect(res.body[0].password).toBe(userTestData[0].password)
+            expect(res.body[0].preferredPaymentMethod).toBe(userTestData[0].preferredPaymentMethod)
+        })
+
+    })
+    describe('POST route', () =>{
+        it('should add a new user to the database', async () =>{
+            const res = await request
+                .post('/api/users')
+                .send({
+                    uuid: '4',
+                    userName: 'testD',
+                    password: 'testPassD',
+                    preferredPaymentMethod: 'CASH'
+                })
+
+            expect(res.status).toBe(200)
+            expect(res.body.uuid).toBe('4')
+            expect(res.body.userName).toBe('testD')
+            expect(res.body.password).toBe('testPassD')
+            expect(res.body.preferredPaymentMethod).toBe('CASH')
+        })
+        it('should automatically add the default attribute for \'preferredPaymentMethod\' if omitted', async ()=>{
+            const res = await request
+                .post('/api/users')
+                .send({
+                    uuid: '5',
+                    userName: 'testE',
+                    password: 'testPassE'
+                })
+
+            expect(res.body.preferredPaymentMethod).toBe('CASH')
+        })
+        // //ATM UUID is passed in with making a post request, but should change this so a UUID is generated for the user
+        it('should throw an error if uuid, username, password is omitted', () =>{
+                try{  
+                    const res = request
+                    .post('/api/users')
+                    .send({
+                        uuid: '6',
+                        password: 'testPassF'
+                    })
+                }
+                catch(err){
+                    expect(err.message).toBe('userName required')
+                }
+        })
+        it('should not take in any extra information past User Schema', async () =>{
+            const res = await request
+                .post('/api/users')
+                .send({
+                    uuid: '5',
+                    userName: 'testE',
+                    password: 'testPassE',
+                    extra: 'extra_content'
+                })
+
+            expect(res.body.extra).toBe(undefined)
```

Comment:
So does the user get saved when passing an extra info?

Reply:
The only info that passes through is 'uuid', 'userName', 'password', and 'preferredPaymentMethod' when it's passed in to the post body

Reply:
? So does that mean the record doesn't get saved when you try to create a User with the extra info? It's a little unclear what is supposed to happen.

Reply:
if there's missing info from the aforementioned attributes, it'll throw an error. But all the info is present, even if there's additional information, it'll only use the required information and process it into the DB

Reply:
I can adjust the test some more to show that this works

Reply:
Gotcha. From this test, it's not clear whether the record is actually saved or not; just that the payload we get back doesn't contain the extra attribute. Would make sense to query for the record and check whether `extra` is on the record or not.

Reply:
Made the changes for all the POST tests besides the one we're expecting an error from!